mriqc/interfaces/diffusion.py

```diff
@@ -545,3 +654,108 @@ def _extract_b0(in_file, b0_ixs, out_path=None):
+        The data array to be thresholded.
+    z_threshold : float, optional (default=3.0)
+        The number of standard deviations to use above the mean as the threshold
+        multiplier.
+    grouping_vals : ndarray (int, same shape as data or 3D), optional
+        If provided, this array is used to group voxels for thresholding. Voxels
+        with the same value in `grouping_vals` are considered to belong to the same
+        group. The threshold will be calculated independently for each group.
+        - If `grouping_vals` has the same shape as `data` (4D), it is assumed to be
+          a mask where each voxel value indicates the group it belongs to.
+        - If `grouping_vals` has a 3D shape, it is assumed to represent b-values
+          corresponding to each voxel in the 4D `data` array. In this case, voxels
+          with the same b-value are grouped together.
+    bmag : int, optional
+        The order of magnitude for b-value rounding (used only if
+        `grouping_vals` is provided as b-values). Default: None (derived from max b-value).
+
+    Returns:
+    -------
+    ndarray (bool, same shape as data)
+        A binary mask where True values indicate voxels classified as spikes and
+        False values indicate non-spikes. The mask has the same shape as the input
+        data array.
+
+    """
+    from dipy.core.gradients import unique_bvals_magnitude
+    from dipy.core.gradients import round_bvals
+
+    if grouping_vals is None:
+        threshold = np.round((z_threshold * np.std(data)) + np.mean(data), 3)
+        spike_mask = np.round(data, 3) > threshold
+        return spike_mask
+
+    threshold_mask = np.zeros(data.shape)
+
+    rounded_grouping_vals = round_bvals(grouping_vals, bmag)
+    gvals = unique_bvals_magnitude(grouping_vals, bmag)
+
+    if grouping_vals.shape == data.shape:
+        for gval in gvals:
+            gval_data = data[rounded_grouping_vals == gval]
+            gval_threshold = ((z_threshold * np.std(gval_data))
+                              + np.mean(gval_data))
+            threshold_mask[rounded_grouping_vals == gval] = (
+                gval_threshold * np.ones(gval_data.shape))
+    else:
+        for gval in gvals:
+            gval_data = data[..., rounded_grouping_vals == gval]
+            gval_threshold = ((z_threshold * np.std(gval_data))
+                              + np.mean(gval_data))
+            threshold_mask[..., rounded_grouping_vals == gval] = (
+                gval_threshold * np.ones(gval_data.shape))
+
+    spike_mask = data > threshold_mask
+
+    return spike_mask
+
+
+def noise_b0(data, gtab, mask=None):
+    """
+    Estimate noise in raw dMRI based on b0 variance.
+
+    Parameters
+    ----------
+    """
+    if mask is None:
+        mask = np.ones(data.shape[:3], dtype=bool)
+    b0 = data[..., ~gtab.b0s_mask]
+    return np.percentile(np.var(b0[mask], -1), (25, 50, 75))
+
+# OE: Create interface to mask the CC
```

Comment:
@arokem Same as above. We already fit a DTI model, so we should use the FA from that step (unless this one has something that makes it special). I think, again, that the CC mask can be really helpful for future IQM definitions beyond those we are adding at this time. Also worth plotting, I think.